sip-test-kit/src/main/java/de/ikor/sip/foundation/testkit/workflow/whenphase/routeinvoker/impl/FtpRouteInvoker.java

```diff
@@ -0,0 +1,129 @@
+package de.ikor.sip.foundation.testkit.workflow.whenphase.routeinvoker.impl;
+
+import static de.ikor.sip.foundation.testkit.workflow.whenphase.routeinvoker.headers.FtpExchangeHeaders.*;
+import static org.apache.camel.Exchange.*;
+
+import de.ikor.sip.foundation.testkit.util.SIPExchangeHelper;
+import de.ikor.sip.foundation.testkit.workflow.givenphase.Mock;
+import de.ikor.sip.foundation.testkit.workflow.whenphase.routeinvoker.RouteInvoker;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.camel.CamelContext;
+import org.apache.camel.Endpoint;
+import org.apache.camel.Exchange;
+import org.apache.camel.Route;
+import org.apache.camel.component.file.remote.*;
+import org.apache.camel.util.FileUtil;
+import org.apache.camel.util.ObjectHelper;
+import org.apache.commons.net.ftp.FTPFile;
+import org.springframework.stereotype.Component;
+
+/** Invoker class for triggering routes with Ftp, Ftps, Sftp consumers */
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class FtpRouteInvoker implements RouteInvoker {
```

Comment:
Does it make sense to extend the FileInvoker and make the header functions protected there? It seems to me that it's behaving very similar. Maybe invoke function can be templated.

Reply:
Actually it doesn't, header calculations are not the same. Only headers "CamelFileLastModified" and "CamelMessageTimestamp" are calculated in same way, hence invokers have only one same method named "prepareOtherHeaders".